src/main/webapp/WEB-INF/tags/instructor/studentList.tag

```diff
@@ -100,42 +100,74 @@
                                     <c:out value="${student.studentEmail}"/>
                                 </td>
                                 <td class="no-print align-center">
-                                    <a class="btn btn-default btn-xs student-view-for-test"
+                                    <c:set var="viewButtonEnabled" value="${section.allowedToViewStudentInSection}" />
+                                    <a class="btn btn-default btn-xs student-view-for-test<c:if test="${not viewButtonEnabled}"> disabled btn-enable-mouse-events</c:if>"
+                                       <c:choose>
+                                       <c:when test="${not viewButtonEnabled}">
+                                          title="<%= Const.Tooltips.ACTION_NOT_ALLOWED %>"
+                                          disabled
+                                       </c:when>
+                                       <c:otherwise>
+                                          title="<%= Const.Tooltips.COURSE_STUDENT_DETAILS %>"
+                                       </c:otherwise>
+                                       </c:choose>
                                        href="${student.courseStudentDetailsLink}"
-                                       title="<%= Const.Tooltips.COURSE_STUDENT_DETAILS %>"
                                        target="_blank"
                                        data-toggle="tooltip"
-                                       data-placement="top"
-                                       <c:if test="${not section.allowedToViewStudentInSection}">disabled</c:if>>
+                                       data-placement="top">
                                         View
                                     </a>
-                                    <a class="btn btn-default btn-xs student-edit-for-test"
+                                    <c:set var="editButtonEnabled" value="${section.allowedToModifyStudent}" />
+                                    <a class="btn btn-default btn-xs student-edit-for-test<c:if test="${not editButtonEnabled}"> disabled btn-enable-mouse-events</c:if>"
+                                       <c:choose>
+                                       <c:when test="${not editButtonEnabled}">
+                                          title="<%= Const.Tooltips.ACTION_NOT_ALLOWED %>"
+                                          disabled
+                                       </c:when>
+                                       <c:otherwise>
+                                          title="<%= Const.Tooltips.COURSE_STUDENT_EDIT %>"
+                                       </c:otherwise>
+                                       </c:choose>
                                        href="${student.courseStudentEditLink}"
```

Comment:
Since we are doing a choose-otherwise, do you think it'll be better to hide the actual href (and stuff like `target="_blank"`) when the button is disabled